cli/src/commands/evolog.rs

```diff
@@ -151,14 +161,35 @@ pub(crate) fn cmd_evolog(
     if !args.no_graph {
         let mut raw_output = formatter.raw()?;
         let mut graph = get_graphlog(graph_style, raw_output.as_mut());
-        for commit in commits {
-            let edges = commit
-                .predecessor_ids()
+
+        let commit_dag: Vec<Result<GraphNode<Commit>, Infallible>> = commits
```

Comment:
We don't need `Result<..>` here, right? It's only required by `reverse_graph()`.

Reply:
If this is what you're getting at: Yes, the code in `evolog` doesn't care about the `Result` but it's required by the signature of `reverse_graph`.

From skimming the other uses of `reverse_graph` it seems possible to remove the requirement for a `Result` in the signature but it would require more refactoring changes across several unrelated files. So, perhaps that's another good candidate for a follow-up if we'd like to change that.